.github/workflows/utils/pages_teardown.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+# Pages Teardown
+#
+# Deletes all preview or production branches in a Cloudflare Pages project.
+# Requires the following command line arguments delimited by spaces, as
+# described in the variable names below.
+
+PAGES_PROJECT_NAME=$1
+PAGES_BRANCH=$2
+CF_ACCOUNT_ID=$3
+CF_API_TOKEN=$4
+
+# Hit Cloudflare GET Deployments Route
+res=$(
+  curl --request GET \
+    --url https://api.cloudflare.com/client/v4/accounts/$CF_ACCOUNT_ID/pages/projects/$PAGES_PROJECT_NAME/deployments \
+    --header "Content-Type: application/json" \
+    --header "Authorization: Bearer ${CF_API_TOKEN}"
+)
+
+deployments_length=$(echo "${res}" | jq '.result | length')
+
+# Loop over each deployment
+for ((i=0; i<$deployments_length; i++)); do
+
+  deployment_branch=$(echo "${res}" | jq -c ".result[$i].deployment_trigger.metadata.branch")
```

Comment:
Would be a good idea to check for edge cases on this script. What happens if a deployment exists that wasn't triggered by wrangler (I believe branch can be null)?

Reply:
Tried to find edge cases by doing a "direct upload". Didn't cause any issues, marking as resolved for now.